### PR TITLE
AI Filter for entries

### PR DESCRIPTION
**Pull request by apolignano**
This PR adds an AI filter to the Browse page for InterPro and MemberDBs entries.

- The filter follows the same logic of the GOTerm filter.
- It's not shown if there's no LLM generated entries for a MemberDB (e.g. Pfam)

### CODE REVIEW

File: src/components/Entry/EntryListFilters/index.tsx [hunk 1]
```diff
@@ -8,20 +8,26 @@ import EntryTypeFilter from './EntryTypeFilter';
...
+    <FiltersPanel>
+      <EntryTypeFilter
+        label={`${
+          mainDB === 'InterPro' ? 'InterPro' : 'Member Database Entry'
+        } Type`}
+      />
+      {mainDB !== 'InterPro' && <IntegratedFilter label="InterPro State" />}
+      {mainDB === 'InterPro' && <GOTermsFilter label="GO Terms" />}
+      {mainDB === 'InterPro' && <LatestFilter label="New entries" />}
+      <AIGeneratedFilter label={AILabel} changeLabel={setAILabel} />
```
Comment (matthiasblum) [inline src/components/Entry/EntryListFilters/index.tsx:27]: Unfortunately, changing the label to an empty string only hide the list of options, but there are still an instance of `FilterPanel` being created resulting in an empty space at the bottom:

image

The simplest is too always display the filter, regardless of the whether there are AI signatures or not.

File: src/components/Entry/EntryListFilters/AIGeneratedFilter/index.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  if (!loading) {
+    terms.unshift(['All', NaN]);
+  }
+
+  const [showFilter, setShowFilter] = useState(true);
+
+  useEffect(() => {
+    if (terms.length > 0) {
+      // If loaded AI-generated reviewed and unreviewed counts will always be in position 2 and 3 of the terms obj
+      if (terms[2][1] == 0 && terms[3][1] == 0) {
```
Comment (matthiasblum) [inline src/components/Entry/EntryListFilters/AIGeneratedFilter/index.tsx:69]: Are you sure about that?
`terms[0]` is always `['All', NaN]`, but other items are sorted by number of entries/signatures in the category. It's true that _at the moment_ there are always more curated entries than AI-generated, but you should not assume that it always be the case. 

You should check using the term, e.g.

```ts
if (terms.some(([term, count]) => (term === "AI-Generated (reviewed)" || term === "AI-Generated (unreviewed)") && count > 0)) {
  setShowFilter(true);
} else {
  setShowFilter(false);
}
```

File: src/components/Entry/EntryListFilters/AIGeneratedFilter/index.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+      } else {
+        setShowFilter(true);
+        changeLabel('AI-Generated Entries');
+      }
+    }
+  });
+
+  return (
+    showFilter && (
+      <div className={css('list-go', { stale: isStale })}>
```
Comment (matthiasblum) [inline src/components/Entry/EntryListFilters/AIGeneratedFilter/index.tsx:81]:
```suggestion
      <div className={css('list-curation-status', { stale: isStale })}>
```

**Review by matthiasblum — CHANGES_REQUESTED** [review]
Please see individual suggestions/comments.

File: src/components/Entry/EntryListFilters/index.tsx (hunk 1 above)
Comment (apolignano) [reply src/components/Entry/EntryListFilters/index.tsx:27]: Yeah, I noticed. Unfortunately the way that page is structured seemed to not allow conditional rendering of the filter. Will look better into it to see if there's a better solution.

File: src/components/Entry/EntryListFilters/AIGeneratedFilter/index.tsx (hunk 2 above)
Comment (apolignano) [reply src/components/Entry/EntryListFilters/AIGeneratedFilter/index.tsx:69]: Missed the sorting part. Makes sense now. I'll fix this.

**Comment by apolignano** [comment]
Now employing same logic of the other filters:

- If a filter count is 0, don't show it.
- Show count next to the "All" term


image

**Review by matthiasblum — APPROVED** [review]
LGTM. Feel free to merge.

**Pull request merged**